app/lib/shared/handler_helpers.dart

```diff
@@ -200,23 +203,42 @@ shelf.Handler _sanitizeRequestWrapper(shelf.Handler handler) {
   };
 }
 
-/// Looks at request and if the 'Authorization' header was set tries to get
-/// the user email address and registers it.
+/// Processes `Authorization` header and the session cookie, and on successful
+/// verification it will set the authenticated user and the user session data.
 shelf.Handler _userAuthWrapper(shelf.Handler handler) {
   return (shelf.Request request) async {
+    User user;
     final authorization = request.headers['authorization'];
     if (authorization != null) {
       final parts = authorization.split(' ');
       if (parts.length == 2 && parts.first.trim().toLowerCase() == 'bearer') {
         final accessToken = parts.last.trim();
 
-        final user =
-            await accountBackend.authenticateWithBearerToken(accessToken);
-        if (user != null) {
-          registerAuthenticatedUser(user);
-        }
+        user = await accountBackend.authenticateWithBearerToken(accessToken);
       }
     }
+
+    UserSessionData userSessionData;
+    final cookies =
+        parseCookieHeader(request.headers[HttpHeaders.cookieHeader]);
+    final sessionId = cookies['pub_sid'];
+    if (sessionId != null) {
+      userSessionData = await accountBackend.lookupSession(sessionId);
+    }
+
+    // verify if the ids match
+    if (user != null &&
```

Comment:
I think it would be fine to omit this check.

In fact I think the two middlewares that does session lookup and authorization header check could be independent.

But I'm also not entirely sure, I mean this is fine, it just seems overkill.. Why would the API care about the session.
Let's talk tomorrow..

Reply:
Okay, if you are okay with splitting, I'll move these to separate wrappers and omit the check.